app/src/main/java/com/epfl/beatlink/repository/profile/ProfileRepositoryFirestore.kt

```diff
@@ -10,60 +11,80 @@ import kotlinx.coroutines.tasks.await
     return try {
-      val snapshot = db.collection("userProfiles").document(userId).get().await()
-      // Manually retrieve fields from the snapshot
-      val bio = snapshot.getString("bio")
-      val links = snapshot.getLong("links")?.toInt() ?: 0
-      val name = snapshot.getString("name")
+      val snapshot = db.collection(collectionPath).document(userId).get().await()
       val profilePictureUrl = snapshot.getString("profilePicture")
-      val username = snapshot.getString("username") ?: ""
-      val favoriteMusicGenres = snapshot.get("favoriteMusicGenres") as? List<String> ?: emptyList()
-
-      // Parse profilePicture field to Uri, if it's not null
-      val profilePicture = profilePictureUrl?.let { Uri.parse(it) }
-
-      // Construct the ProfileData instance
-      val data =
+      val profileData =
           ProfileData(
-              bio = bio,
-              links = links,
-              name = name,
-              profilePicture = profilePicture,
-              username = username,
-              favoriteMusicGenres = favoriteMusicGenres)
-      Log.d("PROFILE_FETCH", "Fetched profile data: $data") // Added log for debugging
-      data
+              bio = snapshot.getString("bio"),
+              links = snapshot.getLong("links")?.toInt() ?: 0,
+              name = snapshot.getString("name"),
+              profilePicture = profilePictureUrl?.let { Uri.parse(it) },
+              username = snapshot.getString("username") ?: "",
+              favoriteMusicGenres =
+                  snapshot.get("favoriteMusicGenres") as? List<String> ?: emptyList())
+      Log.d("PROFILE_FETCH", "Fetched profile data: $profileData")
+      profileData
     } catch (e: Exception) {
-      e.printStackTrace()
-      Log.e("PROFILE_FETCH_ERROR", "Error fetching profile: ${e.message}") // Log error details
+      Log.e("PROFILE_FETCH_ERROR", "Error fetching profile: ${e.message}")
       null
     }
   }
 
+  override suspend fun addProfile(userId: String, profileData: ProfileData): Boolean {
+    return try {
+      db.collection(collectionPath).document(userId).set(profileData).await()
+      Log.d("PROFILE_ADD", "Profile added successfully for user: $userId")
+      true
+    } catch (e: Exception) {
+      Log.e("PROFILE_ADD_ERROR", "Error adding profile: ${e.message}")
+      false
+    }
+  }
+
   override suspend fun updateProfile(userId: String, profileData: ProfileData): Boolean {
     return try {
-      db.collection("userProfiles").document(userId).set(profileData).await()
+      db.collection(collectionPath).document(userId).set(profileData).await()
+      Log.d("PROFILE_UPDATE", "Profile updated successfully for user: $userId")
       true
     } catch (e: Exception) {
-      e.printStackTrace()
-      Log.e("PROFILE_UPDATE_ERROR", "Error updating profile: ${e.message}") // Log error details
+      Log.e("PROFILE_UPDATE_ERROR", "Error updating profile: ${e.message}")
+      false
+    }
+  }
```

Comment:
Here the two functions do the exact same thing, meanwhile this is great to provide semantic clarity, it could be improved to avoid duplicated code and rename the function `saveProfile` for example.

Reply:
I’m not entirely sure I understand the suggestion. I followed a similar pattern from the bootcamp, using addProfile and updateProfile for semantic clarity: addProfile is intended for creating a new user profile, while updateProfile handles changes to an existing profile. While they perform similar actions, their distinction reflects their specific use cases.

Reply:
I suggest considering combining the two functions into a single saveProfile function, as it might avoid the duplicated code. However, this is not a critical change — it really depends on the desired organization and the vision for maintaining clarity in the code. Leaving it as-is is perfectly fine too!